server/src/main/java/org/opensearch/action/admin/cluster/allocation/ClusterAllocationExplanation.java

```diff
@@ -95,7 +95,7 @@ public ClusterAllocationExplanation(StreamInput in) throws IOException {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         shardRouting.writeTo(out);
-        out.writeOptionalWriteable(currentNode);
+        out.writeOptionalWriteable((stream, node) -> node.writeToWithAttribute(stream), currentNode);
         out.writeOptionalWriteable(relocationTargetNode);
         out.writeOptionalWriteable(clusterInfo);
         shardAllocationDecision.writeTo(out);
```

Comment:
Do we need to handle versioning for BWC?

Reply:
While skipping writing of attribute, we write the size of attribute as 0. This will automatically take care of backward compatiblity issue as while reading the attribute in previous version, it will first read the size of attribute. And since size of attribute is 0, it wont read any further attribute (threby skipping populating attribute information in DiscoveryNode).